playbooks/prepare_stack_testconfig.yaml

```diff
@@ -0,0 +1,92 @@
+---
+- hosts: standalone
+  become: true
+  become_user: stack
+  gather_facts: false
+  vars_files: vars/defaults.yaml
+
+  tasks:
+
+  - name: Gather host environment variables
+    setup:
+      filter: ansible_env
+
+  - name: Ensure remote .ssh directory
+    file:
+      path: "{{ ansible_env.HOME }}/.ssh"
+      state: directory
+      owner: stack
+      group: stack
+      mode: '0700'
+
+  - name: Copy ssh public key to host
+    copy:
+      src: "{{ testconfig_public_key }}"
+      dest: "{{ ansible_env.HOME }}/.ssh/{{ testconfig_public_key | basename }}"
+      owner: stack
+      group: stack
+      mode: '0600'
+    register: testconfig_public_key_copy
+
+  - name: Configure resources for openshift user
+    shell: |
+      set -e -o pipefail
+
+      if ! openstack keypair show default >/dev/null; then
+          openstack keypair create --public-key {{ testconfig_public_key_copy.path }} default
+      fi
+      if ! openstack network show private >/dev/null; then
+          openstack network create --internal private
+      fi
+      if ! openstack subnet show private-subnet >/dev/null; then
+          openstack subnet create private-subnet --subnet-range {{ testconfig_private_cidr }} --network private
+      fi
+      # create basic security group to allow ssh/ping/dns
+      if ! openstack security group show basic >/dev/null; then
```

Comment:
In all of this openstack show calls you can probably just execute with a flag for the show command. 
I guess it makes no difference, but it lowers need traffic back and forth on the client

Reply:
Also I see a few calls to openstack that the ansible cloud module would exectue as well. 
Maybe I am missing the context of where this is executed. ( 
For example this whole shell command could be broken down into ansible calls, unless it is being executed on a remote machine (removed twice from ansible-playbook is being executed).